src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -700,32 +700,71 @@ public void ExecuteCodeBlockActions<TLanguageKindEnum>(
             CancellationToken cancellationToken)
             where TLanguageKindEnum : struct
         {
-            ExecuteBlockActionsCore<CodeBlockStartAnalyzerAction<TLanguageKindEnum>, CodeBlockAnalyzerAction, SyntaxNodeAnalyzerAction<TLanguageKindEnum>, SyntaxNode, TLanguageKindEnum>(
-                codeBlockStartActions, codeBlockActions, codeBlockEndActions, analyzer,
-                declaredNode, declaredSymbol, executableCodeBlocks, (codeBlocks) => codeBlocks.SelectMany(
-                    cb =>
+            Debug.Assert(!executableCodeBlocks.IsEmpty);
+            var executableNodeActions = ArrayBuilder<SyntaxNodeAnalyzerAction<TLanguageKindEnum>>.GetInstance();
```

Comment:
moved this up here so that we can infer the type arguments to ExecuteBlockActionsCore from just the args passed in.